
package encfs;

option optimize_for = CODE_SIZE;

import "interface.proto";

message EncfsConfig
{
    optional string creator = 1;
    optional string writer = 11;
    optional int32 revision = 2 [default=0];

    required Interface cipher = 3;
    required EncryptedKey key = 4;
    
    optional Interface naming = 5;
    optional bool unique_iv = 51 [default=false];
    optional bool chained_iv = 52 [default=false];
    optional bool external_iv = 53 [default=false];

    required int32 block_size = 6;
    optional int32 block_mac_bytes = 61 [default=0];
    optional int32 block_mac_rand_bytes = 611 [default=0];
    optional bool allow_holes = 62 [default = false];

}

message EncryptedKey
{
    required bytes ciphertext = 1;
    required int32 size = 2;    // Size of data in bytes

    optional bytes salt = 6;

    optional int32 kdf_iterations = 10;
    optional int32 kdf_duration = 11 [default=500];
}

